File: test_data/typescript/test_main/aas_core_meta.v3/input/snippets/Verification/ID_shorts_of_variables_are_unique.ts
/**
 * Check that all {@link types.IReferable.idShort}'s are unique
 * among values of `inputVariables`, `outputVariables`
 * and `inoutputVariables`.
 *
 * @param inputVariables - to be verified
 * @param outputVariables - to be verified
 * @param inoutputVariables - to be verified
 * @returns `true` if the check passes
 */
export function idShortsOfVariablesAreUnique(
  inputVariables: Iterable<AasTypes.OperationVariable> | null,
  outputVariables: Iterable<AasTypes.OperationVariable> | null,
  inoutputVariables: Iterable<AasTypes.OperationVariable> | null,
): boolean {
  const idShortSet = new Set<string>();

  if (inputVariables !== null) {
    for (const variable of inputVariables) {
      if (variable.value.idShort !== null) {
        if (idShortSet.has(variable.value.idShort)) {
          return false;
        }

        idShortSet.add(variable.value.idShort);
      }
    }
  }

  if (outputVariables !== null) {
    for (const variable of outputVariables) {
      if (variable.value.idShort !== null) {
        if (idShortSet.has(variable.value.idShort)) {
          return false;
        }

        idShortSet.add(variable.value.idShort);
      }
    }
  }

  if (inoutputVariables !== null) {
    for (const variable of inoutputVariables) {
      if (variable.value.idShort !== null) {
        if (idShortSet.has(variable.value.idShort)) {
          return false;
        }

        idShortSet.add(variable.value.idShort);
      }
    }
  }

  return true;
}

File: test_data/java/test_main/aas_core_meta.v3/input/snippets/Verification/ID_shorts_of_variables_are_unique.java
/**
* Check that all {@link IReferable#getIdShort idShort} 's are among all the inputVariables, outputVariables and inoutputVariables are unique.
* @param inputVariables the inputVariables
 * @param outputVariables the outputVariables
 * @param inoutputVariables the inoutputVariables
*/
public static boolean idShortsOfVariablesAreUnique(
    Iterable<? extends IOperationVariable> inputVariables,
    Iterable<? extends IOperationVariable> outputVariables,
    Iterable<? extends IOperationVariable> inoutputVariables) {

  Set<String> idShortSet = new HashSet<>();

  if (inputVariables != null) {
    for(IOperationVariable variable : inputVariables) {
      if (variable.getValue().getIdShort().isPresent()) {
        if (idShortSet.contains(variable.getValue().getIdShort().get())) {
          return false;
        }
        idShortSet.add(variable.getValue().getIdShort().get());
      }
    }
  }

  if (outputVariables != null) {
    for (IOperationVariable variable : outputVariables) {
      if (variable.getValue().getIdShort().isPresent()) {
        if (idShortSet.contains(variable.getValue().getIdShort().get())) {
          return false;
        }
        idShortSet.add(variable.getValue().getIdShort().get());
      }
    }
  }

  if (inoutputVariables != null) {
    for (IOperationVariable variable :  inoutputVariables) {
      if (variable.getValue().getIdShort().isPresent()) {
        if (idShortSet.contains(variable.getValue().getIdShort().get())) {
          return false;
        }
        idShortSet.add(variable.getValue().getIdShort().get());
      }
    }
  }

  return true;
}
